validation/immersed_boundaries/immersed_bickley_jet.jl

```diff
@@ -0,0 +1,256 @@
+#####
+
+Ψ(y) = - tanh(y)
+U(y) = sech(y)^2
+
+# A sinusoidal tracer
+C(y, L) = sin(2π * y / L)
+
+# Slightly off-center vortical perturbations
+ψ̃(x, y, ℓ, k) = exp(-(y + ℓ/10)^2 / 2ℓ^2) * cos(k * x) * cos(k * y)
+
+# Vortical velocity fields (ũ, ṽ) = (-∂_y, +∂_x) ψ̃
+ũ(x, y, ℓ, k) = + ψ̃(x, y, ℓ, k) * (k * tan(k * y) + y / ℓ^2) 
+ṽ(x, y, ℓ, k) = - ψ̃(x, y, ℓ, k) * k * tan(k * x) 
+
+"""
+    run_bickley_jet(output_time_interval = 2, stop_time = 200, arch = CPU(), Nh = 64, ν = 0,
+                    momentum_advection = VectorInvariant())
+
+Run the Bickley jet validation experiment until `stop_time` using `momentum_advection`
+scheme or formulation, with horizontal resolution `Nh`, viscosity `ν`, on `arch`itecture.
+"""
+function run_bickley_jet(; output_time_interval = 2, stop_time = 200, arch = CPU(), Nh = 64, ν = 0, advection = WENO5())
+
+    # Regular model
+    grid = RegularRectilinearGrid(size=(Nh, Nh), halo=(3, 3),
+                                  x = (-2π, 2π), y=(-2π, 2π),
+                                  topology = (Periodic, Bounded, Flat))
+
+    regular_model = IncompressibleModel(architecture = arch,
+                                        advection = advection,
+                                        timestepper = :RungeKutta3,
+                                        grid = grid,
+                                        tracers = :c,
+                                        closure = IsotropicDiffusivity(ν=ν, κ=ν),
+                                        coriolis = nothing,
+                                        buoyancy = nothing)
+
+    # Non-regular model
+    solid(x, y, z) = y > 2π
+
+    expanded_grid = RegularRectilinearGrid(size=(Nh, Int(5Nh/4)), halo=(3, 3),
+                                           x = (-2π, 2π), y=(-2π, 3π),
+                                           topology = (Periodic, Bounded, Flat))
+
+    immersed_grid = ImmersedBoundaryGrid(expanded_grid, GridFittedBoundary(solid))
+
+    immersed_model = IncompressibleModel(architecture = arch,
+                                         advection = advection,
+                                         timestepper = :RungeKutta3,
+                                         grid = immersed_grid,
+                                         tracers = :c,
+                                         closure = IsotropicDiffusivity(ν=ν, κ=ν),
+                                         coriolis = nothing,
+                                         buoyancy = nothing)
+
+    # ** Initial conditions **
+    #
+    # u, v: Large-scale jet + vortical perturbations
+    #    c: Sinusoid
+
+    # Parameters
+    ϵ = 0.1 # perturbation magnitude
+    ℓ = 0.5 # Gaussian width
+    k = 0.5 # Sinusoidal wavenumber
+
+    # Total initial conditions
+    uᵢ(x, y, z) = U(y) + ϵ * ũ(x, y, ℓ, k)
+    vᵢ(x, y, z) = ϵ * ṽ(x, y, ℓ, k)
+    cᵢ(x, y, z) = C(y, grid.Ly)
+
+    set!(regular_model, u=uᵢ, v=vᵢ, c=cᵢ)
+    set!(immersed_model, u=uᵢ, v=vᵢ, c=cᵢ)
```

Comment:
```suggestion
    set!(immersed_model, u=uᵢ, v=vᵢ, c=cᵢ, mass=1)
```

Reply:
I know in this model it doesn't matter as much, but to simulate the wall accurately it might be better to set the initial condition to be zero within the solid, just to be consistent for other IBM's of the future.

Reply:
We are applying the mask in `set!` right now:

https://github.com/CliMA/Oceananigans.jl/blob/1775ebb75bfc6beeef8b3c147223588af4488326/src/Models/IncompressibleModels/set_incompressible_model.jl#L44-L47

so there's no need to require this of users. Do you think this is a good approach / convenient for users?

Feel free to commit my suggested changes (or reject them!) plus all of the awesome code you've written for the analysis (which doesn't appear to be part of the PR yet?)

Reply:
Gotcha okay that makes sense. It is probably easier for users not to have to deal with that! And I haven't added it yet, because it was just in a notebook and not a validation script until today!